Docs: add WordCamp contributor day page (#528)
WordPress/wordpress-playground: packages/docs/site/docs/14-contributor-day/01-index.md

```
npm install -g @wp-now/wp-now
```

### Running

To start a new server, navigate to your WordPress plugin or theme directory and run the command wp-now start. This will start the server with the default PHP and WordPress versions. A new tab will open automatically in your browser.

```
cd gutenberg
wp-now start
```

You can stop wp-now by killing the process by pressing the keys `CTRL + C`.

To change the PHP or WordPress version, you can use the `--php` and `--wp` arguments. For example, to start a server with PHP 7.4 and WordPress 6.1, you can run:

```
wp-now start --php 7.4 --wp 6.1
```

## Contributing to Gutenberg

### Create a Gutenberg PR Step By Step

Begin by creating a fork of the Gutenberg repository using your GitHub account. Once you've done that, clone your forked repository to your local machine to download the code. Following this, install the necessary dependencies and build the Gutenberg files.

```
git clone git@github.com:WordPress/gutenberg.git
cd gutenberg
npm install
npm run dev
```

Ensure that wp-now is installed on your system. Then, open a new terminal window and start the WordPress server by executing the wp-now command.

```
cd gutenberg

> wp-now start

Reviewer: One issue I just ran into...

My shell runs `nvm use` automatically, so I was switched to `node 14` when I opened Gutenberg:

<img width="1014" alt="image" src="https://github.com/WordPress/wordpress-playground/assets/36432/c6d47fd8-b2ae-436e-ba5f-89994cbaf702">

Reviewer 2: Wasn't there some `wp-now` code to display a node version warning in this case?

For more in detail information, check the block editor handbook: https://developer.wordpress.org/block-editor/contributors/code/getting-started-with-code-contribution/#getting-the-gutenberg-code

### Test a Gutenberg PR Step By Step

To test a Gutenberg PR with wp-now, start by navigating to the PR on the Gutenberg repository. Once there, checkout the PR branch to switch your local environment to the version of the code that includes the proposed changes. After checking out the branch, pull the latest changes to ensure your local copy is up to date. Next, run npm install to ensure all dependencies are up to date. This step is crucial as it ensures that your testing environment matches the one in which the changes were made. Finally, test the changes in your local environment. This could involve running unit tests and manually testing features in a browser.

```
# copy the branch-name from GitHub
git checkout branch-name
git pull
npm install
npm run dev

# In a different terminal inside the gutenberg folder
wp-now start
```

### Test a Gutenberg PR with the web version of WordPress Playground

> You can manually test a Gutenberg pull request directly from your browser. Given a pull request URL, https://github.com/WordPress/gutenberg/pull/51239, copy the ID (in this case, `51239`) and paste it into the following link: https://playground.wordpress.net/?gutenberg-pr=51239. Use your specific pull request ID. Open this Playground link in your browser to initiate a new WordPress playground where you can manually review and validate the changes proposed in the pull request.

Reviewer 2: ```suggestion
You can manually test a Gutenberg pull request directly from your browser. Given a pull request URL, https://github.com/WordPress/gutenberg/pull/51239, copy the ID (in this case, `51239`) and paste it at https://playground.wordpress.net/gutenberg.html. Use your specific pull request ID. Once you press "Go," your Pull Request will open in a new WordPress Playground where you can review the proposed changes!
```

> ![Mermaid chart](https://github.com/WordPress/wordpress-playground/assets/779993/d4bc5a27-1401-4f35-bc68-2fad0ef699d5)

Reviewer 2: @sejas For the future – Docusaurus can render mermaid charts directly, there's no need to screenshot them:

https://docusaurus.io/docs/next/markdown-features/diagrams
